### PR TITLE
feat: 🎸 [JIRA: HCPSDKFIORIUIKIT-2708] avatar stack support

### PR DESCRIPTION
**Pull request by xiaoyu0722**
|grouped|horizontal|
|-|-|
|Simulator Screenshot - iPhone 15 Pro Max - 2024-09-03 at 17 04 52|Simulator Screenshot - iPhone 15 Pro Max - 2024-09-03 at 17 05 00|

### CODE REVIEW

File: Apps/Examples/Examples/FioriSwiftUICore/AvatarStack/AvatarStackExample.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            }
+        }
+        .avatarsLayout(self.model.avatarsLayout)
+        .isAvatarCircular(self.model.isCircular)
+        .avatarsTitlePosition(self.model.titlePosition)
+        .avatarsSpacing(self.model.spacing)
+        .avatarsMaxCount(self.model.maxCount)
+        .avatarBorderColor(self.model.borderColor)
+        .avatarBorderWidth(self.model.borderWidth)
+        .avatarSize(self.avatarSize)
```
Comment (billzhou0223) [inline Apps/Examples/Examples/FioriSwiftUICore/AvatarStack/AvatarStackExample.swift:29]: Environment is useful when you have multiple UI elements of the same type in the hierarchy and you want to be able to set something in the parent and propagate down to all the children. In your case, I don't see this type of requirement.

Let's try to make some of the properties as init params. The others can be configured through style modifier.

```
AvatarStack(layout: .grouped, spacing: 8, ...)
      .avatarsStyle {
             Avatars($0)
                  .borderColor()
                  .borderWidth()
      }
```

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Apps/Examples/Examples/FioriSwiftUICore/AvatarStack/AvatarStackExample.swift (hunk 1 above)
Comment (xiaoyu0722) [reply Apps/Examples/Examples/FioriSwiftUICore/AvatarStack/AvatarStackExample.swift:29]: These environments are for uniform settings of avatars.
`.avatarsStyle {}` is for the whole `avatars` which is new grouped view generated by `AvatarBuilder`.

If developer want to set different styles for different avatars, they should do the styles setting in the avatars initialization like below way.

Screenshot 2024-09-05 at 14 39 45

Comment (billzhou0223) [reply Apps/Examples/Examples/FioriSwiftUICore/AvatarStack/AvatarStackExample.swift:29]: OK let's keep them as environment values. But at lease we can combine border width and color into one api. Same as SwiftUI's border modifier.

Comment (xiaoyu0722) [reply Apps/Examples/Examples/FioriSwiftUICore/AvatarStack/AvatarStackExample.swift]: Updated to `func avatarsBorder(_ color: Color, width: CGFloat = 1) -> some View`

**Review by billzhou0223 — APPROVED** [review]

**Pull request merged**